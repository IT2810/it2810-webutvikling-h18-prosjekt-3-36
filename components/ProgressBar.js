import React from 'react';
import {
    StyleSheet, View, Text
  } from 'react-native';
import Colors from '../constants/Colors';

export default class ProgressBar extends React.Component {
  render() {
    return (
      <View style={styles.outerBar}>
          <View style={{backgroundColor: Colors.tintColor, borderRadius: 5, height: 15, width: this.props.percentage + "%"}}>
          </View>
      </View>
    );
  }
}
const styles = StyleSheet.create({
    outerBar: {
      backgroundColor: "gray",
      borderRadius: 5,
      width: "92%",
      margin: "4%"
    }
  });
