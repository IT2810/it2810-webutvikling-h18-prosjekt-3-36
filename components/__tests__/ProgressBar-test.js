import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import ProgressBar from "../ProgressBar";
import appState from '../../store/reducers/Reducer'
import { createStore } from 'redux';

const store = createStore(appState)

it("ProgressBar Matches snapshot.", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<ProgressBar store={store} navigation={{ getParam: () => { }, setParams: () => { } }} isModalVisible={false} />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
});