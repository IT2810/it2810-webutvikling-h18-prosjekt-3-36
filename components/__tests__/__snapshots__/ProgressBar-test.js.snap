// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Matches snapshot. 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "gray",
      "borderRadius": 5,
      "margin": "4%",
      "width": "92%",
    }
  }
>
  <Component
    style={
      Object {
        "backgroundColor": "#ff2d55",
        "borderRadius": 5,
        "height": 15,
        "width": "undefined%",
      }
    }
  />
</Component>
`;
